### DISCUSSION TITLE
The CV jobs can not finished.

### DISCUSSION THREAD

**Discussion by linwanhsing** [ttadano/alamode#212] [Q&A]
Dear Dr. Tadano,

As I take the Cross validation calculation by running the different sets of CV calculations separately, several hours later, each job always suspend at "CV = -1: Manual CV mode.   Validation data is read from DFSET_CV" in the log file.

Could you please help me to solve this problem? 

Thanks.

**Some details as below.**

Program ALM  Ver.  1.2.0
Number of OpenMP threads = 1

**more abc_alm_cvset1.in** 
&general
  PREFIX = abc222_cvset1
  MODE = optimize
  NAT = 48; NKD = 3
  KD =a b c
/

&interaction
  NORDER = 3  # up to quartic
  NBODY = 2 3 3 
/

&cutoff 
  *-* None 10 10 10
/

&optimize
  LMODEL = enet
  NDATA = 100
  DFSET = DFSET_randomQ
  FC2XML = ../2_generate_config/abc222_harmonic.xml

  CV = -1  # manual CV
  L1_RATIO = 1.0 # LASSO
  CV_MINALPHA = 1.0e-4
  CV_MAXALPHA = 0.2
  CV_NALPHA = 30

  STANDARDIZE = 1
  CONV_TOL = 1.0e-4

  SKIP = 1-25
  NSTART_CV = 1
  NEND_CV = 25
/

&cell
     1.89 # factor in Bohr unit
...

**tail -20 abc_alm_cvset1.log**
  25 entries will be used for validation.

  Total Number of Parameters : 22239
  Total Number of Free Parameters : 2652

Elastic-net  cross-validation with the following parameters:
   L1_RATIO = 1
   CV =              -1
   CV_MINALPHA =          0.0001  CV_MAXALPHA =             0.2
   CV_NALPHA =    30
   CONV_TOL =          0.0001
   MAXITER = 10000
   ENET_DNORM =               1

  STANDARDIZE = 1 : Standardization will be performed for matrix A and vector b.
                    The ENET_DNORM-tag will be neglected.

  CV = -1: Manual CV mode.
           Validation data is read from DFSET_CV {This is the last line}.

**Comment by ttadano** [comment]
I recommend that you upgrade ALAMODE to the latest version. If the problem still occurs in v. 1.5.0, please let us know.

**Reply by linwanhsing** [reply]
Dear Dr. Tadano,

 Thanks for your answer. While the same problem still occurs in v. 1.5.0 for both way as below.

**1. the AIMD method for CV**
   **$ more abc_alm_cv.in** 
&general
  PREFIX = abc222
  MODE = optimize
  NAT = 48; NKD = 3
  KD = a b c
/


&interaction
  NORDER = 3  # 1: harmonic, 2: cubic, ..
  NBODY = 2 3 3
/

&cutoff 
  *-* None 15.0 9.0
/

&optimize
  LMODEL = enet
  NDATA = 80
  DFSET = DFSET_AIMD_random
  FC2XML = abc222_harmonic.xml

  CV = 4 # split the DFSET to 4 sets and perform CV
  L1_RATIO = 1.0 # LASSO
  CV_MINALPHA = 1.0e-8
  CV_MAXALPHA = 0.01
  CV_NALPHA = 30

  STANDARDIZE = 1
  CONV_TOL = 1.0e-8

  ! STOP_CRITERION = 30
/

&cell
  1.89 # factor in Bohr unit
...

**$ more abc_alm_cv.log**

  Program ALM
  Ver.  1.5.0          

 Number of OpenMP threads = 1
...

**$ tail abc_alm_cv.log** 

  STANDARDIZE = 1 : Standardization will be performed for matrix A and vector b.
                    The ENET_DNORM-tag will be neglected.

  CV > 0: Automatic CV mode.

  Start 4-fold CV with 80 Datasets


  SET :   1 (**the last line**)

**2. the random method for CV** 
**$ more abc_alm_cvset1.in** 
&general
  PREFIX = abc222_cvset1
  MODE = optimize
  NAT = 48; NKD = 3
  KD = a b c
/


&interaction
  NORDER = 3  # up to quartic
  NBODY = 2 3 3 
/

&cutoff 
  *-* None None 10.0
/

&optimize
  LMODEL = enet
  NDATA = 100
  DFSET = DFSET_randomQ
  FC2XML = abc222_harmonic.xml

  CV = -1  # manual CV
  L1_RATIO = 1.0 # LASSO
  CV_MINALPHA = 1.0e-8
  CV_MAXALPHA = 0.2
  CV_NALPHA = 30

  STANDARDIZE = 1
  CONV_TOL = 1.0e-5

  SKIP = 1-25
  NSTART_CV = 1
  NEND_CV = 25
/

&cell
     1.89 # factor in Bohr unit
...

**$ more abc_alm_cvset1.log**
 Program ALM
 Ver.  1.5.0 

 Number of OpenMP threads = 1


**$ tail abc_alm_cvset1.log**
   MAXITER = 10000
   STOP_CRITERION =     5
   ENET_DNORM =               1

  STANDARDIZE = 1 : Standardization will be performed for matrix A and vector b.
                    The ENET_DNORM-tag will be neglected.

  CV = -1: Manual CV mode.
           Validation data is read from DFSET_CV (**the last line**)

The regular POTCAR used in the VASP calculations.

However, the **7.6_anharm_IFCs/3_cv/si_alm_cvset2.in** can be finished by ALAMODE/1.2.0.

**Reply by ttadano** [reply]
Thank you for the report.

Unfortunately, it is difficult to identify the origin of the issue. That could be a bug in the code or a large memory requirement. If you don't mind, please send the input files to @ttadano via email to investigate the issue.

**Reply by ttadano** [reply]
After some investigations using the input files obtained from @linwanhsing, the above issue has been resolved by setting `FCSYM_BASIS = C`.

If you face a similar issue related to `'boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<std::out_of_range> >'`, please try `FCSYM_BASIS = C`.

**Reply by linwanhsing** [reply]
Dear Dr. Tadano,

The Cross validation calculation of the AIMD random method has been finished with your helpful suggestion, and the finite temperture phonon dispersion obtained.

Thanks a lot.

**Comment by liudadie12138** [comment]
@ttadano Dear Dr. Tadano, I tried to use phonopy to calculate the NaNbO3 phonon spectrum, using the expansion of 331 to calculate and did not find the virtual frequency, and then tried to use alamode to calculate, I used the 221 expansion when the phonon spectrum was calculated and the phonopy calculation result was very similar, but there was an imaginary frequency, and then I tried to use alamode to calculate the 331 expansion cell, this time the calculation result not only did not eliminate the virtual frequency, but the calculation result and phonopy appeared relatively different, Why is this and is there any solution to it?

**Reply by ttadano** [reply]
Please make a seperate discussion.
